README.md

[<img src="https://github-ads.s3.eu-central-1.amazonaws.com/support-ukraine.svg?t=1" />](https://supportukrainenow.org)

# This is my package test-levenshtein

[![Latest Version on Packagist](https://img.shields.io/packagist/v/getcodedev/test-levenshtein.svg?style=flat-square)](https://packagist.org/packages/getcodedev/test-levenshtein)
[![GitHub Tests Action Status](https://img.shields.io/github/workflow/status/getcodedev/test-levenshtein/run-tests?label=tests)](https://github.com/getcodedev/test-levenshtein/actions?query=workflow%3Arun-tests+branch%3Amain)
[![GitHub Code Style Action Status](https://img.shields.io/github/workflow/status/getcodedev/test-levenshtein/Check%20&%20fix%20styling?label=code%20style)](https://github.com/getcodedev/test-levenshtein/actions?query=workflow%3A"Check+%26+fix+styling"+branch%3Amain)
[![Total Downloads](https://img.shields.io/packagist/dt/getcodedev/test-levenshtein.svg?style=flat-square)](https://packagist.org/packages/getcodedev/test-levenshtein)

This is where your description should go. Limit it to a paragraph or two. Consider adding a small example.

## Installation

You can install the package via composer:

```bash
composer require getcodedev/test-levenshtein
```

You can publish and run the migrations with:

```bash
php artisan vendor:publish --tag="test-levenshtein-migrations"
php artisan migrate
```

## Usage

```php
$duplicates = check_duplicates(
    model: User::class,
    search: [
        'first_name' => 'John',
        'home' => [
            'address' => "bbbbb, rrr BB, 77777",
        ],
    ],
    concat_search_columns: [
        'home' => [
            'address' => 'homes.street, homes.city homes.state, homes.zip',
        ],
    ],
    priority_columns: [
        'homes.address',
        'users.first_name',
    ],
    with_similarity_min_common: 50,
    limit: 10
);
```

## Testing

```bash

# 1. clone this package && run composer install
composer install

# 2. Then need to set configs for mysql connection in phpunit.xml
cp phpunit.xml.dist phpunit.xml

# 3. Run tests
composer test
```

## Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information on what has changed recently.

## Contributing

Please see [CONTRIBUTING](https://github.com/spatie/.github/blob/main/CONTRIBUTING.md) for details.

## Security Vulnerabilities

Please review [our security policy](../../security/policy) on how to report security vulnerabilities.

## Credits

- [Roman Nebesnuy](https://github.com/GetCodeDev)
- [All Contributors](../../contributors)

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All notable changes to `test-levenshtein` will be documented in this file.


src/Commands/SeedCommand.php
<?php

namespace GetCodeDev\TestLevenshtein\Commands;

use GetCodeDev\TestLevenshtein\Models\Home;
use GetCodeDev\TestLevenshtein\Models\Test;
use GetCodeDev\TestLevenshtein\Models\User;
use Illuminate\Console\Command;

class SeedCommand extends Command
{
    public $signature = 'seed';

    public $description = 'My command';

    public function handle(): int
    {
        //$this->runTestSeeder();
        //$this->testCheck();

        $street = '61946 Elena Skyway Apt. 654';
        $city = 'East Gwen';
        $state = 'ND';
        $zip = 46786;

        $user_correct = User::factory()
            ->state([
                'first_name' => 'Ricky123',
                'last_name' => 'Smith123',
            ])
            ->for(Home::factory()->state([
                'street' => $street,
                'city'   => $city,
                'state'  => $state,
                'zip'    => $zip,
            ]))
            ->create();

        $user_not_correct = User::factory()
            ->state([
                'first_name' => 'Erna',
                'last_name' => 'Kilback',
            ])
            ->for(Home::factory()->state([
                'street' => '1111',
                'city'   => '222',
                'state'  => 'AA',
                'zip'    => 44444,
            ]))
            ->create();

        $check = check_duplicates(
            model: User::class,
            search: [
                'home' => [
                    'address' => "bbbbb, rrr BB, 77777",
                ],
            ],
            concat_search_columns: [
                'home' => [
                    'address' => 'homes.street, homes.city homes.state, homes.zip',
                ],
            ],
            priority_columns: [
                'homes.address',
            ],
            with_similarity_min_common: 50
        );

        dd($check);

        $this->comment('All done');

        return self::SUCCESS;
    }

    protected function runTestSeeder(): void
    {
        User::factory()
            ->count(100)
            ->hasJobs(3)
            ->create();
    }

    protected function testCheck()
    {
        $first_name = 'Ricky';
        $last_name = 'Rogers';

        $street = '61946 Elena Skyway Apt. 654';
        $city = 'East Gwen';
        $state = 'ND';
        $zip = 46786;

        $concat_search_columns = [
            'users' => [
                'full_name' => 'users.last_name users.first_name',
            ],
        ];

        $a = check_duplicates(
            model: User::class,
            search: [
                'first_name' => $first_name,
                'last_name'  => $last_name,
                'full_name'  => $first_name.' '.$last_name,
                'home' => [
                    'street' => $street,
                ],
                'jobs' => [
                    'street' => $street,
                ],
                //'test' => true
            ],
            //concat_search_columns: $concat_search_columns,
            priority_columns: [
                'full_name',
                'first_name',
                'last_name',
                'homes.street',
                'jobs.street',
            ],
            limit: 5,
        //with_similarity_min_common: 50
        );

        dd($a);
    }
}
